planner/functional_dependency/fd_graph.go

```diff
@@ -33,6 +33,32 @@ func (s *FDSet) closureOf(colSet FastIntSet) FastIntSet {
 			// If the closure is updated, we redo from the beginning.
 			i = -1
 		}
+		// why this? it is caused by our definition of equivalence as {superset} == {superset},
+		// which is also seen as {superset} --> {superset}. but when we compute the transitive
+		// closure, `fd.from.SubsetOf(resultSet)` is not transitive here. Actually, we can also
+		// see the equivalence as {single element} == {superset} / {single element} --> {superset}.
+		if fd.equiv && fd.from.ElementSubsetOf(resultSet) && !fd.to.SubsetOf(resultSet) {
```

Comment:
We can change the `fd.from.ElementSubsetOf(resultSet)` to `fd.From.Intersects(resultSet)`?